bazel/cython_library.bzl

```diff
@@ -72,7 +72,10 @@ def pyx_library(name, deps = [], py_deps = [], srcs = [], **kwargs):
         native.cc_binary(
             name = shared_object_name,
             srcs = [stem + ".cpp"],
-            deps = deps + ["@local_config_python//:python_headers"],
+            deps = deps + [
+            "@rules_python//python/cc:current_py_cc_headers",
+            "@rules_python//python/cc:current_py_cc_libs",
```

Comment:
You probably don't want py_cc_libs here -- that is going to try and link in the libpython.so files. The previous code, if its name is to be believed, just gets the headers (which is what you want -- lazy binding of the runtime symbols to come from whatever the py runtime the extension is used with).

Ah, actually, looking at variety.tpl, I think what it was doing is more nuanced. If I'm reading that variety.tpl code correctly, `:python_headers` was just the headers for non-windows, but the headers and cc_import(<libs>) for windows. I think the equivalent here would be:

```
deps = [current_py_cc_headers] + select({
"windows": [current py cc libs],
"default: []
})
```

This probably isn't _the_ problem, but would manifest as the sort of weird behavior ODR violates can result in.

Reply:
Indeed originally, I only depended on the headers but pulled in the patch from https://github.com/bazelbuild/bazel-central-registry/pull/2685 later.

I used your `select()` suggestion. Let's see what the tests are doing with this.

Reply:
Reading the original bug more closely, it appeared on MacOS, so only linking against it on Windows probably won't be sufficient. I reverted the `select()` again but registered a hermetic toolchain as per the docs to replace the previous custom `python_configure()`.